contracts/BalanceTrackerFixedPrice.sol

```diff
@@ -0,0 +1,332 @@
+    }
+
+    // Check and record delivery rate
+    function checkAndRecordDeliveryRate(
+        address mech,
+        address requester,
+        uint256 requestId,
+        bytes memory paymentData
+    ) external payable {
+        // Check for marketplace access
+        if (msg.sender != mechMarketplace) {
+            revert ManagerOnly(msg.sender, mechMarketplace);
+        }
+
+        // Get mech max delivery rate
+        uint256 maxDeliveryRate = IMech(mech).maxDeliveryRate();
+
+        // Get payment token
+        address token;
+        if (paymentData.length == 32) {
+            // Extract token address
+            token = abi.decode(paymentData, (address));
+            if (token != address(0) && msg.value > 0) {
+                revert NoDepositAllowed(msg.value);
+            }
+        } else if (paymentData.length > 0) {
+            revert InvalidPayloadLength(paymentData.length, 32);
+        }
+
+        // Get account balance
+        uint256 balance = mapRequesterBalances[requester][token];
+
+        // Check the request delivery rate for a fixed price
+        if (balance < maxDeliveryRate) {
+            revert InsufficientBalance(balance, maxDeliveryRate);
+        }
+
+        // Record request token
+        mapRequestIdTokens[requestId] = token;
+
+        // Adjust account balance
+        balance -= maxDeliveryRate;
+        mapRequesterBalances[requester][token] = balance;
+    }
+
+    /// @dev Finalizes mech delivery rate based on requested and actual ones.
+    /// @param mech Delivery mech address.
+    /// @param requester Requester address.
+    /// @param requestId Request Id.
+    /// @param maxDeliveryRate Requested max delivery rate.
+    function finalizeDeliveryRate(address mech, address requester, uint256 requestId, uint256 maxDeliveryRate) external {
+        // Check for marketplace access
+        if (msg.sender != mechMarketplace) {
+            revert ManagerOnly(msg.sender, mechMarketplace);
+        }
+
+        // Get actual delivery rate
+        uint256 actualDeliveryRate = IMech(mech).getFinalizedDeliveryRate(requestId);
+
+        // Check for zero value
+        if (actualDeliveryRate == 0) {
+            revert ZeroValue();
+        }
+
+        // Get token associated with request Id
+        address token = mapRequestIdTokens[requestId];
+
+        uint256 rateDiff;
+        if (maxDeliveryRate > actualDeliveryRate) {
+            // Return back requester overpayment debit
+            rateDiff = maxDeliveryRate - actualDeliveryRate;
+            mapRequesterBalances[requester][token] += rateDiff;
+        } else {
+            actualDeliveryRate = maxDeliveryRate;
+        }
+
+        // Record payment into mech balance
+        mapMechBalances[mech][token] += actualDeliveryRate;
+
+        emit MechPaymentCalculated(mech, requestId, actualDeliveryRate, rateDiff);
+    }
+
+    // TODO buyBackBurner does not account for other tokens but WETH, OLAS
```

Comment:
We should for fixed price only use the native token or OLAS imo, so this is not an issue

if in the future other tokens are needed we cross that bridge then